app/actions/remote/role.ts

```diff
@@ -42,12 +42,13 @@ export const fetchRolesIfNeeded = async (serverUrl: string, updatedRoles: string
 
     try {
         const roles = await client.getRolesByNames(newRoles);
-
-        if (roles.length) {
-            await operator.handleRole({
-                roles,
-                prepareRecordsOnly: false,
-            });
+        if (!fetchOnly) {
+            if (roles.length) {
```

Comment:
We can have these two conditions in one single if.

Reply:
done.